src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -52,18 +69,37 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
 
-        // Check the user has the company_incorporation=create permission
-        boolean hasCompanyIncorporationCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_INCORPORATION, Value.CREATE);
+        boolean isRoeUpdateJourneyRequest = isRequestRoeUpdate(request);
 
         var authInfoMap = new HashMap<String, Object>();
         authInfoMap.put(TRANSACTION_ID_KEY, transactionId);
         authInfoMap.put("request_method", request.getMethod());
-        authInfoMap.put("has_company_incorporation_create_permission", hasCompanyIncorporationCreatePermission);
+        authInfoMap.put("is_roe_update_journey_request", isRoeUpdateJourneyRequest);
 
-        if (hasCompanyIncorporationCreatePermission) {
-            ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_incorporation=create permission",
-                    authInfoMap);
-            return true;
+        if (isRoeUpdateJourneyRequest) {
+            boolean doCompanyNumbersMatch = doCompanyNumbersMatch(request, response, transactionId);
+            // Check the user has the company_oe_annual_update=create permission for ROE Update
+            boolean hasCompanyOeAnnualUpdateCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_OE_ANNUAL_UPDATE, Value.CREATE);
+            authInfoMap.put("has_company_oe_annual_update_create_permission", hasCompanyOeAnnualUpdateCreatePermission);
+            authInfoMap.put("roe_update_journey_company_numbers_match", doCompanyNumbersMatch);
+            if (hasCompanyOeAnnualUpdateCreatePermission && doCompanyNumbersMatch) {
+                ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_oe_annual_update=create permission",
+                        authInfoMap);
+                return true;
+            } else {
+                ApiLogger.errorContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey ", null, authInfoMap);
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
```

Comment:
In the scenario that the call to get the transaction has resulted in an exception, `null` is returned from the `getCompanyNumberInTransaction` method and a 500 status code has been set on the response I think this will be overridden at this point to be SC_UNAUTHORIZED, which is incorrect.

Please could you add a unit test to prove/disprove this theory and rework logic (if required). Thanks.